src/python/WMCore/ReqMgr/DataStructs/Request.py

```diff
@@ -20,6 +20,9 @@
 from WMCore.ReqMgr.DataStructs.RequestStatus import REQUEST_START_STATE
 from WMCore.ReqMgr.DataStructs.RequestError import InvalidSpecParameterValue
 
+#TODO: this need to be update from DB instead of hard coding for more flexibility
+RESULT_MASK_FOR_DAS = ["RequestName", "InputDatasets", "PrepID", "Campaign"]
+
```

Comment:
Sure, by scanning one workload_cache, I'd say we need to provide to DAS:
["RequestName", "PrepID", "CMSSWVersion", "GlobalTag"]

not sure about (since people can look at reqmgr to get this info):
["MCPileup", "Memory", "ScramArch", "TimePerEvent", "SizePerEvent"]

I removed "Campaign" because it's encoded in the dataset name (which should be the same as AcquisitionEra in the end).

As I said during the meeting, some fields don't need to be a list (should not). But we need to properly find these keys in a TaskChain or StepChain request:
PrepID, CMSSWVersion, GlobalTag (and Memory, ScramArch, TimePerEvent, SizePerEvent)